File: ShapeByte/features/workout/history/data/WorkoutHistoryDataSource.swift
//
//  WorkoutHistoryDataSource.swift
//  ShapeByte
//
//  Created by Lang, Stefan [RTL Tech] on 04.08.24.
//

import Foundation
import Combine

protocol WorkoutHistoryDataSource {
    func historyFor(date: Date, untilPastDate pastDate: Date) -> [WorkoutScheduleEntry]
}

File: features/workout/data/contract/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/data/contract/history/WorkoutHistoryDataSource.kt
package de.stefan.lang.shapebyte.features.workout.data.contract.history

import de.stefan.lang.foundation.core.contract.loadstate.LoadState
import de.stefan.lang.shapebyte.features.workout.data.contract.schedule.WorkoutScheduleEntry
import kotlin.time.Instant

public interface WorkoutHistoryDataSource {
    public suspend fun historyForDates(
        date: Instant,
        pastDate: Instant,
    ): LoadState.Result<List<WorkoutScheduleEntry>>
}
